web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -845,7 +849,80 @@ private static void warnLegacyTicketTypes(String ticketType) {
         }
     }
 
-    @JsMethod
+    public Promise<SharedExportBytesUnion> shareObject(ServerObject object, SharedExportBytesUnion sharedTicketBytes) {
+        if (object.getConnection() != this) {
+            return Promise.reject("Cannot share an object that comes from another server instance");
+        }
+        PublishRequest request = new PublishRequest();
+        request.setSourceId(object.typedTicket().getTicket());
+
+        Ticket ticket = sharedTicketFromStringOrBytes(sharedTicketBytes);
+        request.setResultId(ticket);
+
+        return Callbacks.grpcUnaryPromise(c -> {
+            sessionServiceClient().publishFromTicket(request, metadata(), c::apply);
+        }).then(ignore -> Promise.resolve(sharedTicketBytes));
+    }
+
+    private static Ticket sharedTicketFromStringOrBytes(SharedExportBytesUnion sharedTicketBytes) {
+        Ticket ticket = new Ticket();
+        final int length;
+        final TypedArray.SetArrayUnionType array;
+        if (sharedTicketBytes.isString()) {
+            byte[] arr = sharedTicketBytes.toString().getBytes(StandardCharsets.UTF_8);
+            length = arr.length;
+            array = TypedArray.SetArrayUnionType.of(arr);
+        } else {
+            Uint8Array bytes = (Uint8Array) sharedTicketBytes;
+            length = bytes.length;
+            array = TypedArray.SetArrayUnionType.of(bytes);
+        }
+        Uint8Array bytesWithPrefix = new Uint8Array(length + 2);
+        // Add the shared ticket prefix at the start of the provided value
```

Comment:
Please either put the prefix in a constant and/or comment linking it to where this prefix is defined. Not having a link is just going to be a headache for future maintenance, and as an outside observer, this is indeed magic.
```suggestion
        // Add the shared ticket prefix at the start of the provided value
        // Shared ticket prefix is defined in {@link io.deephaven.proto.util.SharedTicketHelper#TICKET_PREFIX}
```